### ISSUE TITLE
Respecting asymmetries in injections

### ISSUE THREAD

**Issue by alecgunny** [ML4GW/aframe#402]
This will probably involve integrations with `ml4gw`, but I'm realizing more and more that our default behavior of centering injections and treating the `trigger_offset` symmetrically probably doesn't make much sense. We've talked before about how centering the raw waveform attaches the left tail of the waveform to the end of the coalescence, which is potentially problematic especially for lower mass injections. I've also been thinking that it makes sense to allow the coalescence to be closer to the right of the kernel than to the left, and that `trigger_offset` should maybe just refer to the latter.

I'm sure there are other places where this is crops up, but just generally it might be worth encoding explicitly the physical fact that the coalescence is not some center point of a symmetric signal but rather its end. Not sure how this might apply to other waveform types (e.g. I can imagine this is maybe not true for sine gaussians).

**Comment by EthanMarx** [comment]
Along these lines, I think we should move towards using [pycbcs API](https://pycbc.org/pycbc/latest/html/waveform.html#) for generating waveforms

**Comment by wbenoit26** [comment]
I like that idea

**Comment by alecgunny** [comment]
Yeah agreed. Plus if they have CuPy bindings we could in principle use [dlpack](https://docs.cupy.dev/en/stable/reference/generated/cupy.ndarray.html#cupy.ndarray.toDlpack) to generate waveforms on the fly on the GPU and zero-copy them to [torch tensors](https://pytorch.org/docs/stable/dlpack.html)

**Comment by alecgunny** [comment]
Actually wow yeah especially as we add containerization it looks like install PyCBC with GPU support is reasonably straightforward (see [here](https://pycbc.org/pycbc/latest/html/install_cuda.html)). Obviously doesn't take priority over the paper, but this could definitely be worth moving high up the list for immediately after.

**Comment by rafia17** [comment]
This simple routine uses PyCBC to generate the waveforms, putting the coalescence at the right edge:

from pycbc.waveform import get_td_waveform

# setting prior values using the existing bilby functions
priors = nonspin_bns()
params = priors.sample(n_samples)
sample_params = [
        dict(zip(params, col)) for col in zip(*params.values())]

waveform_size = int(sample_rate * waveform_duration)

signals = np.zeros([n_samples, num_pols, waveform_size])

for i, p in enumerate(sample_params):
        hp, hc = get_td_waveform(approximant=approximant_bns,   **p)
        hp, hc = hp.trim_zeros(), hc.trim_zeros()
        m     = max(hp.data)
        # find the peak position and then only output the waveform around the peak position
        k     = np.array([int(i) for i,j in enumerate(hp.data) if j==m])
        x_limit_low, x_limit_high = k-65536, k
        signals[i][0] = hp.data[int(x_limit_low):int(x_limit_high)]
        signals[i][1] = hc.data[int(x_limit_low):int(x_limit_high)]
-------------------------------
It was interesting to see that lets say you are generating 10 waveforms, they will all be different lengths, get_td_waveform returns varying lengths of timeseries every time its called depending on the params sent. Lower masses tend to give a longer timeseries. (higher frequencies?). Anyways, in this routine, I took the last 16 sec worth of datapoints including the coalescence, which appears at the right edge

![image](https://github.com/ML4GW/aframe/assets/55510827/1523eead-49d3-4cbc-b851-c4ffb338f2a5)
